src/ambianic/server.py

```diff
@@ -94,40 +93,30 @@ def _configure(env_work_dir=None):
     assert env_work_dir, 'Working directory required.'
     assert os.path.exists(env_work_dir), \
         'working directory invalid: {}'.format(env_work_dir)
-    print("Configuring server...")
-    secrets_file = os.path.join(env_work_dir, SECRETS_FILE)
-    config_file = os.path.join(env_work_dir, CONFIG_FILE)
-    try:
-        if os.path.isfile(secrets_file):
-            with open(secrets_file) as sf:
-                secrets_config = sf.read()
-        else:
-            secrets_config = ""
-            log.warning('Secrets file not found. '
-                        'Proceeding without it: %s',
-                        secrets_file)
-        with open(config_file) as cf:
-            base_config = cf.read()
-            all_config = secrets_config + "\n" + base_config
-        config = yaml.safe_load(all_config)
-        log.debug('loaded config from %r: %r', CONFIG_FILE, config)
+
+    config_manager.stop()
+
+    def logging_config_handler(config):
         # configure logging
         logging_config = None
         if config:
             logging_config = config.get('logging', None)
         _configure_logging(logging_config)
+
+    config_manager.register_handler(logging_config_handler)
```

Comment:
This registers regular app logging handler to react on changes to logging settings in config.yaml.

Reply:
I do not see registering handler for the other sections in config yaml: sources, ai models, and pipelines. Is that left for a future PR? That is fine with me , we just need to document which parts of config.yaml are reactive and have hot reload capability and which ones require app restart.

Reply:
Yes, this PR aims at introducing the reactive configuration capabilities. I would go feature by feature to identify the right approach and expose a corresponding HTTP API.

I can do all in this PR but was thinking to have smaller PR with specific improvements.

Let me know your preferred approach

Reply:
Got it. I also like incremental PRs that are easy to comprehend and absorb. If anything regresses, which it never should if our coverage is good, we can quickly identify the problem, fix and add more test coverage. We can count this PR as phase 1 towards #191